File: src-server/src/db/schemas/agent.py
from __future__ import annotations
from . import DTOBase
from .provider import LlmModelRead

class AgentBase(DTOBase):
    name: str
    system_prompt: str

class AgentBrief(DTOBase):
    id: int
    name: str

class AgentRead(AgentBase):
    id: int
    model: LlmModelRead | None = None

class AgentCreate(AgentBase):
    model_id: int | None = None

class AgentUpdate(DTOBase):
    name: str | None = None
    system_prompt: str | None = None
    model_id: int | None = None

File: src-frontend/src/types/agent.ts
import type { IconName } from "@/features/Tabs/AgentPanel/IconSelectDialog";
import type { LlmModelRead } from "./provider";

export type AgentBase = {
  name: string;
  icon_name: IconName;
  system_prompt: string;
};

export type AgentBrief = {
  id: number;
  name: string;
  icon_name: IconName;
};

export type AgentRead = AgentBase & {
  id: number;
  model: LlmModelRead | null;
};

export type AgentCreate = AgentBase & {
  model_id: number | null;
};

export type AgentUpdate = Partial<AgentCreate>;
